### DISCUSSION TITLE
Setting trace ID based on X-Amzn-Trace-Id

### DISCUSSION THREAD

**Discussion by bbhoss** [cogini/phoenix_container_example#1] [Q&A]
I didn't see anywhere in here that you have a plug to parse the `X-Amzn-Trace-Id` header field and set the trace id in the context from it. I found `opentelemetry_xray_propagator` and it seems like it could handle it, but it's not clear how to use it.

**Comment by bbhoss** [comment]
Update: It seems like potentially this should just be working out of the box, but it's not. I think it's related to Bandit (vs Cowboy) downcasing the headers, it doesn't match the case from `opentelemetry_xray_propagator`. Not sure what the best fix is or if that's even the issue.

**Reply by patmaddox** [reply]
I haven't tried working with this example application, but we found that the config key is now `text_map_propagators`. So the config would be:

```elixir
config :opentelemetry,
  text_map_propagators: [:opentelemetry_xray_propagator, :baggage]
```

The header case doesn't matter, otel library does the right thing.

The other line that needs to be called _somewhere_ as part of the request is:

```elixir
:otel_propagator_text_map.extract(conn.req_headers)
```

which happens as a side effect of [`OpentelemetryPhoenix.setup`](https://github.com/opentelemetry-beam/opentelemetry_phoenix/blob/21bb6483982f8ad9f38ff5a1acf8e1968b4c8a5a/lib/opentelemetry_phoenix.ex#L106). If one of the other libraries isn't calling it, you'll need to add it in a plug.
